### PR TITLE
Remove MapAxis.to_table() from IRF.to_hdulist

### LINKED ISSUE #6624: Remove MapAxis.to_table() from IRF.to_hdulist
MapAxis is used in several objects: Map (and every container using maps) as well as IRF. In many places, MapAxis and MapAxes are responsible for exporting the axis to a proper HDU. This complexifies a lot the code in MapAxis which has to deal with numerous formats and contexts. A symptom of this is the format names that need to be created which mixes format and context (e.g. "gadl-dl3").

Gammapy needs to support various formats and metadata scheme. The IO code must therefore be more easily adaptable. 

The MapAxis.to_table and to_header must therefore be simplified a lot. Object specific serialization must be performed at the object level not down to MapAxis. 

A first cleanup consist in removing `self.axes.to_table()` in `IRF.to_hdulist()` and explicitly do the table formatting there instead. Then one can remove the `gadf-dl3` format part from MapAxis.

### PR DESCRIPTION
**Pull request by MSCarrasco**
This pull request resolves #6624 
self.axes.to_table() was removed from IRF.to_hdulist() and the table formatting is explicitly done there instead. 
The gadf-dl3 format part was removed from MapAxis and MapAxes.

### CODE REVIEW

File: gammapy/irf/tests/test_io.py [hunk 1]
```diff
@@ -138,6 +140,24 @@ def setup_method(self):
...
 
+    def test_to_hdulist(self):
+        hdulist = self.aeff.to_hdulist()
+        assert hdulist[0].name == "PRIMARY"
+        assert hdulist[1].name == "EFFECTIVE AREA"
+        assert hdulist[1].header["TUNIT1"] == "TeV"
+        assert hdulist[1].header["TUNIT3"] == "deg"
+        assert hdulist[1].header["TUNIT5"] == "cm2"
+
+        table = Table.read(hdulist[1], format="fits")
```
Comment (registerrier) [inline gammapy/irf/tests/test_io.py]: I am not sure you need to explicitly create a new Table object here. 
hdulist[1].data should be a table HDU already:
https://docs.astropy.org/en/stable/io/fits/index.html#working-with-table-data

**Review by registerrier — COMMENTED** [review]
Thanks @MSCarrasco . This is an important first step.

It seems that some PSF classes re-implment `to_table` and incorrectly call into MapAxes.to_table().
This suggests that the code could be in a helper function and be called in the two functions that need it.

**Comment by bkhelifi** [comment]
Thanks a lot @MSCarrasco .
I think that the tests could be improved with these missing items:
- `IRF.to_table()` not tested
- Line 519-521 in core.py: The background model edge case (if hduclass in {\"BKG\", \"RAD_MAX\"} and column_prefix == \"ENERG\")
- `EnergyDispersion2D`: No test for to_hdulist() with the new implementation
- `Background3D`: No test for to_hdulist() with the new implementation
- `RadMax2D`: Not tested at all
- Background3D.to_hdulist()` 
- Different axis configurations (center vs edges node types)
- Verification that migra, fov_lon, fov_lat axes are correctly processed
Maybe this can be done into an other PR... I leave the Lead Developpers decide

**Review by bkhelifi — CHANGES_REQUESTED** [review]
Looks OK for me. I leave a comment about the test coverage

File: gammapy/irf/tests/test_io.py (hunk 1 above)
Comment (MSCarrasco) [reply gammapy/irf/tests/test_io.py]: Thanks for pointing this, it is now changed

**Comment by MSCarrasco** [comment]
> Thanks a lot @MSCarrasco . I think that the tests could be improved with these missing items:
> 
>     * `IRF.to_table()` not tested
> 
>     * Line 519-521 in core.py: The background model edge case (if hduclass in {"BKG", "RAD_MAX"} and column_prefix == "ENERG")
> 
>     * `EnergyDispersion2D`: No test for to_hdulist() with the new implementation
> 
>     * `Background3D`: No test for to_hdulist() with the new implementation
> 
>     * `RadMax2D`: Not tested at all
> 
>     * Background3D.to_hdulist()`
> 
>     * Different axis configurations (center vs edges node types)
> 
>     * Verification that migra, fov_lon, fov_lat axes are correctly processed
>       Maybe this can be done into an other PR... I leave the Lead Developpers decide

Hi @bkhelifi, thanks for all these important points. I believe I implemented everything, but let me knwo if it is not the case.
The  Line 519-521 part was simply removed as it had been left by mistake.

@registerrier I implemented the helper function as you suggested.

File: gammapy/irf/tests/test_io.py [hunk 2]
```diff
@@ -137,6 +144,134 @@ def setup_method(self):
...
         )
+        self.rad_max = RadMax2D(
+            axes=[self.energy_axis_true.copy(name="energy"), self.offset_axis],
+            data=np.random.rand(9, 3),
+            unit="deg",
+        )
+
+    def test_aeff_to_hdulist(self):
+        with pytest.raises(ValueError, match="Not a valid supported format: 'gadf'"):
+            hdulist = self.aeff.to_hdulist(format="gadf")
```
Comment (registerrier) [inline gammapy/irf/tests/test_io.py]:
```suggestion
            self.aeff.to_hdulist(format="gadf")
            
```

File: gammapy/irf/tests/test_io.py [hunk 3]
```diff
@@ -137,6 +144,134 @@ def setup_method(self):
...
+        hdu_center = aeff_center.to_table_hdu()
+        assert_allclose(hdu_center.data["THETA_LO"][0], offset_center.center.value)
+        assert_allclose(hdu_center.data["THETA_HI"][0], hdu_center.data["THETA_LO"][0])
+
+    @requires_data()
+    def test_psf_3gauss_to_table(self):
+        filename = "$GAMMAPY_DATA/tests/unbundled/irfs/psf.fits"
+        psf = EnergyDependentMultiGaussPSF.read(filename, hdu="POINT SPREAD FUNCTION")
+        with pytest.raises(ValueError, match="Not a valid supported format: 'gadf'"):
+            table = psf.to_table(format="gadf")
```
Comment (registerrier) [inline gammapy/irf/tests/test_io.py]:
```suggestion
            psf.to_table(format="gadf")
            
```

**Review by registerrier — APPROVED** [review]
Thanks @MSCarrasco . Looks good. 
At some point we will have to deprecate the so-called `gadf-dl3` format. This will come in future PRs.
